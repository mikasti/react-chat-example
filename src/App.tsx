import React from 'react';
import ChatPage from './pages/ChatPage';
import './Assets/CSS/main.scss';

// Root component containing our pages

const App: React.FC = () => {
  const chatPage = <ChatPage />;

  return (
    <div className="main">
      {chatPage}
    </div>
  );
};

export default App;
