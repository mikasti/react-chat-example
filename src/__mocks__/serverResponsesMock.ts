import MakeSubmitUserMessage from '../components/helpers/MakeSubmitUserMessage';
import { IDialog } from '../types/ChatTypes';
import { IUser } from '../types/MainTypes';

export const userProfileMock: IUser = {
  userUID: '007',
  name: 'Michael Ivanov',
  bio: 'Frontend Developer',
  isOnline: true,
  nick: '@michael',
  eMail: 'michael@test.com',
  phone: '+357 77 777 777',
};

export const jamieProfileMock: IUser = {
  userUID: '008',
  name: 'Jamie Olivera',
  bio: 'Support engenier',
  isOnline: true,
  nick: '@jamieoliver',
  eMail: 'jamieoliver@test.com',
  phone: '+357 88 888 888',
};

export const gordonProfileMock: IUser = {
  userUID: '001',
  name: 'Gordon Ramsey',
  bio: 'Head engenier',
  isOnline: true,
  nick: '@gordonramsey',
  eMail: 'gordonramsey@test.com',
  phone: '+357 11 111 111',
};


export const userMessagesMock: IDialog = {
  messages: [
    MakeSubmitUserMessage(jamieProfileMock, 'Good morning! What can I make for you today?'),
    MakeSubmitUserMessage(gordonProfileMock, 'Make a normal breakfast!!!'),
    MakeSubmitUserMessage(jamieProfileMock, 'As usual?'),
    MakeSubmitUserMessage(gordonProfileMock, 'No, omelete, now!!!'),
    MakeSubmitUserMessage(userProfileMock, 'Harsh:('),
  ]
}
