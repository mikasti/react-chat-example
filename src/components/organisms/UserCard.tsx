import React, { useCallback, useContext, useState } from 'react';
import { IUser } from '../../types/MainTypes';
import UserProfile from '../molecules/userCard/UserProfile';
import UserData from '../molecules/userCard/UserData';
import AppContext from '../context/AppContext';
import Button from '../atom/common/Button';
import OnlyUserMessages from '../molecules/userChat/OnlyUserMessages';
import makeThemeClassName from '../helpers/makeThemeClassname';
import '../../assets/css/userCard/user-only-messages.scss';

interface IUserCard {
  profile: IUser,
}

const UserCard: React.FC<IUserCard> = React.memo(({ profile }) => {
  const { isDarkTheme } = useContext(AppContext);
  const [isShowUserMessages, setIsShowUserMessages] = useState<boolean>(false);

  const {
    name, bio, image, isOnline, nick, eMail, phone, userUID,
  } = profile;

  const handleShowUserMessages = () => {
    setIsShowUserMessages(!isShowUserMessages);
  };

  const userOnlyMessages = makeThemeClassName('user-only-messages', isDarkTheme);

  return (
    <>
      <UserProfile
        name={name}
        bio={bio}
        isOnline={isOnline}
        image={image}
      />
      <UserData
        nick={nick}
        eMail={eMail}
        phone={phone}
      />
      <div className={userOnlyMessages}>
        <div className='user-only-messages-button-wrapper'>
          <Button
            className='user-only-messages-button'
            label={(!isShowUserMessages) ? 'Show user messages' : 'Hide user messages'}
            onClick={handleShowUserMessages}
          />
        </div>
        {isShowUserMessages && userUID && <OnlyUserMessages uid={userUID} />}
      </div>

    </>
  );
});

export default UserCard;
