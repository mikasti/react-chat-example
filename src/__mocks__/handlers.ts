import { rest } from 'msw';
import { userProfileMock } from './serverResponsesMock';

export const handlers = [
    rest.post('my.messages.pet.project/profile', (req, res, ctx) => {
        return res(
            ctx.json(userProfileMock),
        )
    }),
]