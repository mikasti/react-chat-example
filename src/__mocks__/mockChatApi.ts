import { userProfileMock, userMessagesMock } from './serverResponsesMock';
import { IUser } from '../types/MainTypes';
import { IDialog, IMessage } from '../types/ChatTypes';

interface IChatApi {
    getMessages: () => Promise<IDialog>,
    getMessagesByUId: (uid: string) => Promise<IDialog>,
    getProfile: () => Promise<IUser>,
}
const responseTimeMs: number = 100;

const mockChatApi: IChatApi = {
    getMessages: async () => {
        return new Promise((resolve) => {
            setTimeout(() => resolve(userMessagesMock), responseTimeMs);
        })
    },
    getMessagesByUId: async (uid: string) => {
        return new Promise((resolve) => {
            const filteredMessages: IMessage[] = userMessagesMock.messages.filter((message) => message.author.userUID === uid);
            setTimeout(() => resolve({ messages: filteredMessages }), responseTimeMs);
        })
    },
    getProfile: async () => {
        return new Promise((resolve) => {
            setTimeout(() => resolve(userProfileMock), responseTimeMs);
        })
    },
};

export default mockChatApi;