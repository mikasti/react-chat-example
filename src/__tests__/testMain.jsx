import React from 'react';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../App';
import UserCard from '../components/organisms/UserCard';
import { userProfileMock } from '../__mocks__/serverResponsesMock';

describe('Render App', () => {
  it('App renders correctly', () => {
    const Component = render(<App />);
    expect(Component).toMatchSnapshot();
  });
});

describe('Render <Molecules>', () => {
  it('UserCard renders correctly', () => {
    const Component = render(<UserCard profile={userProfileMock} />);
    expect(Component).toMatchSnapshot();
  });
  it('Render UserCard and check rendered data', () => {
    render(<UserCard profile={userProfileMock} />);
    const name = screen.getByText(userProfileMock.name);
    const nick = screen.getByText(userProfileMock.nick);
    expect(name && nick).toBeInTheDocument();
  });
});
