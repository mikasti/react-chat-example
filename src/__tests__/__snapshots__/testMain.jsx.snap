// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Molecules> UserCard renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user-profile"
      >
        <div
          class="user-img"
        >
          MI
        </div>
        <div
          class="user-name"
        >
          Michael Ivanov
        </div>
        <div
          class="user-bio"
        >
          Frontend Developer
        </div>
        <button
          class="user-status"
          type="button"
        >
          Online
        </button>
      </div>
      <div
        class="user-data"
      >
        <div
          class="user-info"
        >
          <div
            class="user-info-secondary"
          >
            Nickname
          </div>
          <div
            class="user-info-primary"
          >
            @michael
          </div>
        </div>
        <div
          class="user-info"
        >
          <div
            class="user-info-secondary"
          >
            Email
          </div>
          <div
            class="user-info-primary"
          >
            michael@test.com
          </div>
        </div>
        <div
          class="user-info"
        >
          <div
            class="user-info-secondary"
          >
            Phone
          </div>
          <div
            class="user-info-primary"
          >
            +357 77 777 777
          </div>
        </div>
      </div>
      <div
        class="user-only-messages"
      >
        <button
          class="user-only-messages-button"
          type="button"
        >
          Show user messages
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user-profile"
    >
      <div
        class="user-img"
      >
        MI
      </div>
      <div
        class="user-name"
      >
        Michael Ivanov
      </div>
      <div
        class="user-bio"
      >
        Frontend Developer
      </div>
      <button
        class="user-status"
        type="button"
      >
        Online
      </button>
    </div>
    <div
      class="user-data"
    >
      <div
        class="user-info"
      >
        <div
          class="user-info-secondary"
        >
          Nickname
        </div>
        <div
          class="user-info-primary"
        >
          @michael
        </div>
      </div>
      <div
        class="user-info"
      >
        <div
          class="user-info-secondary"
        >
          Email
        </div>
        <div
          class="user-info-primary"
        >
          michael@test.com
        </div>
      </div>
      <div
        class="user-info"
      >
        <div
          class="user-info-secondary"
        >
          Phone
        </div>
        <div
          class="user-info-primary"
        >
          +357 77 777 777
        </div>
      </div>
    </div>
    <div
      class="user-only-messages"
    >
      <button
        class="user-only-messages-button"
        type="button"
      >
        Show user messages
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render App App renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
      >
        <div
          class="loader-comp"
        >
          <div
            class="lds-ripple"
          >
            <div />
            <div />
          </div>
          <div
            class="loader-text"
          >
            Loading user messages...
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
    >
      <div
        class="loader-comp"
      >
        <div
          class="lds-ripple"
        >
          <div />
          <div />
        </div>
        <div
          class="loader-text"
        >
          Loading user messages...
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
